bot/cogs/monstersurvey.py

```diff
@@ -0,0 +1,178 @@
+import json
+import logging
+import os
+from asyncio import sleep as asleep
+from typing import Optional, Union
+
+from discord import Embed
+from discord.ext import commands
+from discord.ext.commands import Bot, Context
+
+log = logging.getLogger(__name__)
+
+EMOJIS = {
+    'SUCCESS': u'\u2705',
+    'ERROR': u'\u274C'
+}
+
+
+class MonsterSurvey:
+    """
+    Vote for your favorite monster!
+    This command allows users to vote for their favorite listed monster.
+    Users may change their vote, but only their current vote will be counted.
+    """
+
+    def __init__(self, bot: Bot):
+        """Initializes values for the bot to use within the voting commands."""
+        self.bot = bot
+        self.registry_location = os.path.join(os.getcwd(), 'bot', 'resources', 'monstersurvey', 'monstersurvey.json')
+        with open(self.registry_location, 'r') as jason:
+            self.voter_registry = json.load(jason)
+
+    def json_write(self):
+        log.info("Saved Monster Survey Results")
+        with open(self.registry_location, 'w') as jason:
+            json.dump(self.voter_registry, jason, indent=2)
+
+    @commands.group(
+        name='monster',
+        aliases=['ms']
+    )
+    async def monster_group(self, ctx: Context):
+        """
+        The base voting command. If nothing is called, then it will return an embed.
+        """
+
+        if ctx.invoked_subcommand is None:
+            default_embed = Embed(
+                title='Monster Voting',
+                color=0xFF6800,
+                description='Vote for your favorite monster!'
+            )
+            default_embed.add_field(
+                name='.monster show monster_name(optional)',
+                value='Show a specific monster. If none is listed, it will give you an error with valid choices.',
+                inline=False)
+            default_embed.add_field(
+                name='.monster vote monster_name',
+                value='Vote for a specific monster. You get one vote, but can change it at any time.',
+                inline=False
+            )
+            default_embed.add_field(
+                name='.monster leaderboard',
+                value='Which monster has the most votes? This command will tell you.',
+                inline=False
+            )
+            return await ctx.send(embed=default_embed)
+
+    @monster_group.command(
+        name='vote'
+    )
+    async def monster_vote(self, ctx: Context, name: Optional[Union[int, str]] = None):
+        """Casts a vote for a particular monster, or displays a list of monsters that can be voted for
+        if one is not given."""
+        vote_embed = Embed(
+            name='Monster Voting',
+            color=0xFF6800
+        )
+        if isinstance(name, int):
+            name = list(self.voter_registry.keys())[name]
```

Comment:
- This is off by one, since Python indexing starts at 0
- Needs to handle numbers that are out of range & provide feedback
- Should index by the leaderboard, not by the order of the source monster JSON dictionary. The user has no way to know which number corresponds to which monster otherwise.
- User needs some guidance in the voting embed that an integer vote is possible